plugins/airbrake/lib/samson_airbrake/samson_plugin.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+require 'airbrake_tasks'
+
+module SamsonAirbrake
+  class Engine < Rails::Engine
+  end
+
+  class Notification
+    class << self
+      VALID_RAILS_ENV = /^[a-z]+$/
+      SECRET_KEY = 'airbrake_api_key'
+
+      def deliver_for(deploy)
+        return unless DeployGroup.enabled?
+        return unless deploy.project.notify_airbrake
+        return unless deploy.succeeded?
+
+        deploy.stage.deploy_groups.group_by(&:environment).each do |environment, deploy_groups|
+          rails_env = environment.name.downcase
+          next unless rails_env =~ VALID_RAILS_ENV
+          next unless project_api_key = read_secret(deploy.project, deploy_groups, SECRET_KEY)
+
+          with_airbrake_key(project_api_key) do
+            AirbrakeTasks.deploy(
+              rails_env: rails_env,
+              scm_revision: deploy.job.commit,
+              local_username: deploy.user.name
+            )
+          end
+        end
+      end
+
+      private
+
+      def read_secret(project, deploy_groups, key)
+        resolver = Samson::Secrets::KeyResolver.new(project, deploy_groups)
+        full_key = resolver.expand('unused-param', key).map(&:last).first
+        SecretStorage.read_multi([full_key], include_value: true).values.first&.fetch(:value) # read key without raising
+      end
+
+      # not multi-thread safe... this might lead to some samson errors going to other projects / being lost
```

Comment:
If you don't fork, then should create a global variable with the original airbrake API key (set in initializer) and always restore to that original key.  Otherwise there could be race conditions of > 1 project clobbering this key.